### PR TITLE
Handle building and pushing images on Azure Cloud Shell

### PR DESCRIPTION
**Pull request by rakeshvanga**

### CODE REVIEW

File: tools/readme.md [hunk 1]
```diff
@@ -8,25 +8,28 @@ This tool will re-deploy your resources using the assets generated by Azure Dev
...
-Please install the below pre-requisites if using this on your local machine, or alternately, you can just use azure bash [cloudshell](https://shell.azure.com/ ) for a seamless experience.
+Please install the below pre-requisites if using this on your local machine, or alternately, you can just use azure bash [cloudshell](https://shell.azure.com/) for a seamless experience.
 1. [azure cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
 2. [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
 3. [docker](https://docs.docker.com/engine/install/)
 4. curl
 5. gunzip
 6. tar
 
+**Note:** When using Azure Cloud Shell, only Azure Container Registry is available for hosting docker images.
```
Comment (pragyamehta) [inline tools/readme.md:19]: Let's add this to the documentation also

File: tools/readme.md [hunk 2]
```diff
@@ -8,25 +8,28 @@ This tool will re-deploy your resources using the assets generated by Azure Dev
...
+
 ## Use the tool
 ```
 migrate-devspaces-assets.sh -g ResourceGroupName -n AKSName -h ContainerRegistryName  
 Flags:  
     -g Name of resource group of AKS Cluster [required]
     -n Name of AKS Cluster [required]
-    -h Container registry name. Examples: ACR, Docker [required]
+    -h Container registry name. Examples: ACR, Docker [required] (On Azure Cloud Shell only supports ACR)
```
Comment (pragyamehta) [inline tools/readme.md:27]: nit: Azure Cloud Shell only supports hosting images in ACR

**Review by pragyamehta — APPROVED** [review]

File: tools/readme.md [hunk 3]
```diff
@@ -8,25 +8,28 @@ This tool will re-deploy your resources using the assets generated by Azure Dev
 3. If required, deploys an ingress controller to provide a public endpoint to access the service over the internet.
 
 ## Prerequisites
-Please install the below pre-requisites if using this on your local machine, or alternately, you can just use azure bash [cloudshell](https://shell.azure.com/ ) for a seamless experience.
+Please install the below pre-requisites if using this on your local machine, or alternately, you can just use azure bash [cloudshell](https://shell.azure.com/) for a seamless experience.
```
Comment (daniv-msft) [inline tools/readme.md:11]: Nit: please consider replacing `azure bash` (not sure where it's coming from?) by `Azure Cloud Shell`.

File: tools/migrate-devspaces-assets.sh [hunk 4]
```diff
@@ -125,11 +133,25 @@ else
...
+if [ command -v clouddrive &> /dev/null ]; then
+   CLOUDDRIVEHELP=$(clouddrive -h)
+   if [[ $CLOUDDRIVEHELP == *"Azure Cloud Shell"* ]]; then
+      ISAZURESHELL="true"
+      echo "Detected Azure Cloud Shell."
+   fi
+fi
+
+if [[ "$ISAZURESHELL" == "true" ]]; then
+   echo "Building and pushing to Azure Container registry..."
```
Comment (daniv-msft) [inline tools/migrate-devspaces-assets.sh:146]: Would there be a way to enforce that the container registry the user targets is ACR? If it doesn't, what error message do we provide to the user?

File: tools/readme.md (hunk 1 above)
Comment (daniv-msft) [inline tools/readme.md:19]: Could you please clarify why only ACR is possible on Azure Cloud Shell? Is it because of missing dependencies?

Comment (rakeshvanga) [reply tools/readme.md:19]: Azure cloud shell doesn't support running docker daemon. Here is the documentation from Azure Cloud Shell: https://docs.microsoft.com/en-us/azure/cloud-shell/troubleshooting#cannot-run-the-docker-daemon

File: tools/migrate-devspaces-assets.sh (hunk 4 above)
Comment (rakeshvanga) [reply tools/migrate-devspaces-assets.sh:146]: The az acr build task will fail with an error message saying that no azure container registry has been found, if the user has passed in a dockerhub container registry.
Also the docs, readme and the help text is updated to mention this clearly that on azure cloud shell, only ACR is supported.

File: tools/readme.md (hunk 3 above)
Comment (rakeshvanga) [reply tools/readme.md:11]: Sure, I will update it to be more clear.

File: tools/readme.md (hunk 1 above)
Comment (rakeshvanga) [reply tools/readme.md:19]: Yes, once I check it in, I will coordinate with Nick and Gordon to get it updated!

**Review by daniv-msft — APPROVED** [review]
Looks good to me!

**Pull request merged**